OperatorWrap with token ASSIGN too strict for annotations

I have read check documentation: https://checkstyle.sourceforge.io/config_whitespace.html#OperatorWrap
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac OperatorWrapExample.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="OperatorWrap">
      <property name="tokens" value="ASSIGN"/>
    </module>
  </module>
</module>

/var/tmp $ cat OperatorWrapExample.java
@Annotation(example = {
        "foo",
        "bar"
})
public class OperatorWrapExample {

    public String[] array = {
            "foo",
            "bar"
    };
}

@interface Annotation {
    String[] example();
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.1-all.jar -c config.xml OperatorWrapExample.java
Starting audit...
[ERROR] I:\_tmp\_checkstyle\OperatorWrapExample.java:1:21: '=' should be on a new line. [OperatorWrap]
Audit done.
Checkstyle ends with 1 errors.
```

---

In the above example, the array field allows the first element to be on a new line. If the same is done for an annotation, Checkstyle complains. The only way to "fix" this is to put the first element on the same line as the `= {`. I expect the same behaviour in annotations as in other code.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java
@@ -423,12 +423,7 @@ public class OperatorWrapCheck
             result = adjustParens(rightNode, DetailAST::getPreviousSibling);
         }
 
-        // The ARRAY_INIT AST is confusing. It should be
-        // ARRAY_INIT
-        // |--LCURLY (first child)
-        // `--RCURLY (last child)
-        // but there is no LCURLY for ARRAY_INIT
-        if (result.getType() != TokenTypes.ARRAY_INIT) {
+        if (!TokenUtil.isOfType(result, TokenTypes.ARRAY_INIT, TokenTypes.ANNOTATION_ARRAY_INIT)) {
             while (result.getFirstChild() != null) {
                 result = result.getFirstChild();
             }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheckTest.java
@@ -129,6 +129,17 @@ public class OperatorWrapCheckTest
     }
 
     @Test
+    public void testArraysAssign() throws Exception {
+        final String[] expected = {
+            "18:22: " + getCheckMessage(MSG_LINE_NEW, "="),
+            "36:28: " + getCheckMessage(MSG_LINE_NEW, "="),
+        };
+
+        verifyWithInlineConfigParser(
+                getPath("InputOperatorWrapArrayAssign.java"), expected);
+    }
+
+    @Test
     public void testInvalidOption() throws Exception {
 
         try {

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapArrayAssign.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapArrayAssign.java
@@ -1,0 +1,77 @@
+/*
+OperatorWrap
+option = (default)nl
+tokens = ASSIGN
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.operatorwrap;
+
+import java.lang.annotation.Repeatable;
+import java.util.Arrays;
+
+@Annotation1(example = { // ok
+    "foo",
+    "bar"
+})
+@Annotation2(example = // violation
+{
+    "foo",
+    "bar"
+})
+@Nested(
+    nestedAnn = { // ok
+        @Annotation1(example = { // ok
+            "foo",
+            "bar"
+        })
+    }
+)
+public class InputOperatorWrapArrayAssign {
+    public String[] array = { // ok
+        "foo",
+        "bar"
+    };
+    public String[] array2 = // violation
+    {
+        "foo",
+        "bar"
+    };
+}
+
+@Annotation1(
+    value = "1",
+    example = { // ok
+        "foo",
+        "bar"
+    },
+    other = "2"
+)
+class Second {}
+
+@Annotation1(
+    example = { // ok
+        "foo",
+        "bar"
+    },
+    arr = { // ok
+        "1"
+    }
+)
+class Third {}
+
+@interface Annotation1 {
+    String[] example();
+    String value() default "";
+    String other() default "";
+    String[] arr() default "";
+}
+
+@interface Annotation2 {
+    String[] example();
+}
+
+@interface Nested {
+    Annotation1[] nestedAnn();
+}

